gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifyContentsPerExecutor.java

```diff
@@ -62,9 +67,34 @@ public IdentifyContentsPerExecutor(GCParams gcParams) {
             bloomFilterSize);
   }
 
-  protected Function<String, IdentifiedResult> computeExpiredContentsFunc(
+  protected SerializableFunction1<scala.collection.Iterator<String>, scala.collection.Iterator<Row>>
+      getContentRowsFunc(Map<String, ContentBloomFilter> liveContentsBloomFilterMap) {
+    return result -> getContentRows(result, liveContentsBloomFilterMap);
+  }
+
+  /**
+   * convert scala Iterator of reference into scala Iterator of {@link Row}.
+   *
+   * <p>Each reference can produce one or more rows.
+   */
+  private scala.collection.Iterator<Row> getContentRows(
+      scala.collection.Iterator<String> references,
       Map<String, ContentBloomFilter> liveContentsBloomFilterMap) {
-    return reference -> computeExpiredContents(liveContentsBloomFilterMap, reference);
+    List<Row> rows = new ArrayList<>();
+    references.foreach(
+        reference -> {
+          rows.addAll(computeExpiredContents(liveContentsBloomFilterMap, reference));
```

Comment:
TODO: Accumulating all the rows here will bloat up the driver memory. Need to pushdown this iterator to Nessie commit log traversal.

Reply:
Finished the TODO.